pkg/reconciler/ingress/config/istio.go

```diff
@@ -93,7 +93,7 @@ func parseGateways(configMap *corev1.ConfigMap, prefix string) ([]Gateway, error
 	urls := map[string]string{}
 	gatewayNames := []string{}
 	for k, v := range configMap.Data {
-		if !strings.HasPrefix(k, prefix) {
+		if !strings.HasPrefix(k, prefix) || k == prefix {
```

Comment:
Ensure `k[len(prefix):]` won't cause seg fault, and the following `len(parts)` could only be 1 or 2.

Reply:
I am not sure this _will_ cause a index out of bounds, but it will return an empty string.

Reply:
You are right. An empty string will be returned, and `strings.SplitN("", ".", 2)` has a length of 1. To avoid an empty gateway name and keep the following if/else statement simple, maybe it is meaningful to add `k == prefix` condition here.